adapters/oidc/wildfly/wildfly-subsystem/src/main/java/org/keycloak/subsystem/adapter/extension/KeycloakAdapterConfigService.java

```diff
@@ -147,21 +147,8 @@ public void addRedirectRewriteRule(ModelNode operation, ModelNode model) {
         if (!redirectRewritesRules.isDefined()) {
             redirectRewritesRules = new ModelNode();
         }
-
         String redirectRewriteRuleName = redirectRewriteRule(operation);
-        if (!redirectRewriteRuleName.contains(".")) {
-            redirectRewritesRules.get(redirectRewriteRuleName).set(model.get("value").asString());
-        } else {
-            String[] parts = redirectRewriteRuleName.split("\\.");
-            String provider = parts[0];
-            String property = parts[1];
-            ModelNode redirectRewriteRule = redirectRewritesRules.get(provider);
-            if (!redirectRewriteRule.isDefined()) {
-                redirectRewriteRule = new ModelNode();
-            }
-            redirectRewriteRule.get(property).set(model.get("value").asString());
-            redirectRewritesRules.set(provider, redirectRewriteRule);
-        }
```

Comment:
Can the condition and this part be removed?

Reply:
I'm assuming it is safe to remove this based on how the redirect-rewrite-rule is used. (https://github.com/keycloak/keycloak/blob/master/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java#L391). The key is used as a regex to be matched against the path, replacing the matched section with the value. An example here would be 

`        <redirect-rewrite-rule name="^/wsmaster/api/(.*)$">api/$1/</redirect-rewrite-rule>`

The dot is a valid part of a regex expression and the logic above would break it into two segments "^/wsmaster/api/(" and "*)$", which makes no sense. In fact, without this change this example would cause an error when creating the AdapterConfig object as JSON cannot deserialize this into a Map<String, String>, it expects a Map<String, Object> in AdapterConfig.

The logic above seems to have been copied from the credentials handling, where you can have a key format that uses dot as a separator so you can defined different properties for a particular provider. Example, "jwt.client-keystore-file" with value "/tmp/foo.jks" and then "jwt.token-timeout" with value "10". It produces a Map<String, Object> in AdapterConfig, whereas redirect-rewrite-rule produces a Map<String, String>, which further indicates that the configured key and value should be read as they come.

Reply:
Makes sense to me, thanks!